src/aliceVision/camera/Equidistant.hpp

```diff
@@ -9,175 +9,425 @@
   {
-    return focal() * p + principal_point();
+
+    Vec3 X = pose.rotation() * pt;
+
+    const Eigen::Matrix3d& d_X_d_pt = pose.rotation();
+
+    /* Compute angle with optical center */
+    const double len2d = sqrt(X(0) * X(0) + X(1) * X(1));
+    Eigen::Matrix<double, 2, 2> d_len2d_d_X;
+    d_len2d_d_X(0) = X(0) / len2d;
+    d_len2d_d_X(1) = X(1) / len2d;
+    
+    const double angle_Z = std::atan2(len2d, X(2));
+    const double d_angle_Z_d_len2d = X(2) / (len2d*len2d + X(2) * X(2));
+
+    /* Ignore depth component and compute radial angle */
+    const double angle_radial = std::atan2(X(1), X(0));
+
+    Eigen::Matrix<double, 2, 3> d_angles_d_X;
+    d_angles_d_X(0, 0) = - X(1) / (X(0) * X(0) + X(1) * X(1));
+    d_angles_d_X(0, 1) = X(0) / (X(0) * X(0) + X(1) * X(1));
+    d_angles_d_X(0, 2) = 0.0;
+
+    d_angles_d_X(1, 0) = d_angle_Z_d_len2d * d_len2d_d_X(0);
+    d_angles_d_X(1, 1) = d_angle_Z_d_len2d * d_len2d_d_X(1);
+    d_angles_d_X(1, 2) = - len2d / (len2d * len2d + X(2) * X(2));
+
+
+    const double rsensor = std::min(sensorWidth(), sensorHeight());
+    const double rscale = sensorWidth() / std::max(w(), h());
+    const double fmm = _scale_x * rscale;
+    const double fov = rsensor / fmm;
+    const double radius = angle_Z / (0.5 * fov);
+
+    double d_radius_d_angle_Z = 1.0 / (0.5 * fov);
+
+    /* radius = focal * angle_Z */
+    Vec2 P;
+    P(0) = cos(angle_radial) * radius;
+    P(1) = sin(angle_radial) * radius;
+
+    Eigen::Matrix<double, 2, 2> d_P_d_angles;
+    d_P_d_angles(0, 0) = - sin(angle_radial) * radius;
+    d_P_d_angles(0, 1) = cos(angle_radial) * d_radius_d_angle_Z;
+    d_P_d_angles(1, 0) = cos(angle_radial) * radius;
+    d_P_d_angles(1, 1) = sin(angle_radial) * d_radius_d_angle_Z;
+
+    const Vec2 distorted = this->add_disto(P);
+
+    return getDerivativeCam2ImaWrtPoint() * getDerivativeAddDistoWrtPt(P) * d_P_d_angles * d_angles_d_X * d_X_d_pt;
   }
 
-  // Transform a point from the image plane to the camera plane
-  Vec2 ima2cam(const Vec2& p) const override
+  Eigen::Matrix<double, 2, 3> getDerivativeProjectWrtDisto(const geometry::Pose3& pose, const Vec3 & pt)
   {
-    return ( p -  principal_point() ) / focal();
+    
+    const Vec3 X = pose.rotation() * pt;
+
+    /* Compute angle with optical center */
+    const double len2d = sqrt(X(0) * X(0) + X(1) * X(1));
+    const double angle_Z = std::atan2(len2d, X(2));
+  
+    /* Ignore depth component and compute radial angle */
+    const double angle_radial = std::atan2(X(1), X(0));
+
+    const double rsensor = std::min(sensorWidth(), sensorHeight());
+    const double rscale = sensorWidth() / std::max(w(), h());
+    const double fmm = _scale_x * rscale;
+    const double fov = rsensor / fmm;
+    const double radius = angle_Z / (0.5 * fov);
+
+    /* radius = focal * angle_Z */
+    Vec2 P;
+    P(0) = cos(angle_radial) * radius;
+    P(1) = sin(angle_radial) * radius;
+
+    const Vec2 distorted = this->add_disto(P);
```

Comment:
can we remove it as never use it or should that be used in `getDerivativeAddDistoWrtDisto()`?